app/controllers/categories_controller.rb

```diff
@@ -0,0 +1,45 @@
+class CategoriesController < ApplicationController
+  before_action :find_category, only: [:show, :edit, :update]
+
+  def index
+    @categories = Category.all
+    @category = Category.new
+  end
+
+
+  def show
+    @products = @category.products.paginate(page: params[:page], per_page: 12).where(active: true)
+  end
+
+  def edit
+  end
+
+  def create
+    @category = Category.new(category_params)
+    if @category.save
+      redirect_to categories_path
+    else
+      @categories = Category.all
+      render :index
```

Comment:
Should we display a flash message in this case to indicate that the category was not created?